tools/generator/landing-page.js

```diff
@@ -62,201 +240,154 @@ class LandingPageForm extends LitElement {
-      URL.revokeObjectURL(this.marqueeImage.url);
+    if (this.form.marqueeImage?.url && this.form.marqueeImage.url.startsWith('blob:')) {
+      URL.revokeObjectURL(this.form.marqueeImage.url);
     }
-
-    this.marqueeImage = { url: '', name: '' };
-
+    this.form.marqueeImage = { url: '', name: '' };
     const { file } = e.detail;
     if (!file) return;
 
-    this.uploadFile(file)
+    // TODO: update template after page generation to correct path
+    const path = '/tools/page-builder/.prd-template/';
+    this.uploadFile(file, path)
       .then((url) => {
         if (!url) return;
-
-        this.dispatchEvent(new CustomEvent('show-toast', { detail: { type: 'success', message: 'Image Uploaded', timeout: TOAST_TIMEOUT } }));
-        this.updateDocumentMarquee(url);
-        saveSource(DOCUMENT_PATH, this.data.document)
-          .then(() => {
-            this.dispatchEvent(new CustomEvent('show-toast', { detail: { type: 'success', message: 'Document Marquee Updated', timeout: TOAST_TIMEOUT } }));
-          })
-          .catch((error) => {
-            this.dispatchEvent(new CustomEvent('show-toast', { detail: { type: 'error', message: `Failed to save document: ${error.message}` } }));
-          });
+        this.form.marqueeImage = { url, name: file.name };
+        this.dispatchEvent(new CustomEvent('show-toast', { detail: { type: 'success', message: 'Image Uploaded' } }));
       }).catch((error) => {
         this.dispatchEvent(new CustomEvent('show-toast', { detail: { type: 'error', message: `Upload failed: ${error.message}` } }));
       });
+    this.updatePreview();
+    this.saveFormState();
   }
 
   async handleSubmit(e) {
     e.preventDefault();
+    // TODO: Validate form data
+    if (!this.isFormComplete()) {
+      this.dispatchEvent(new CustomEvent('show-toast', { detail: { type: TOAST_TYPES.ERROR, message: 'Please complete all required fields' } }));
+      return;
+    }
+    // TODO: Create page in DA and upload assets, uploadTemplatedText
+    // eslint-disable-next-line no-console
+    console.table(this.form);
+  }
+
+  getSections() {
+    return [
+      () => renderContentType(this.form, this.handleInput),
+      () => renderForm(this.form, this.handleInput, { marketoPOIOptions: this.marketoPOIOptions }),
+      () => renderMarquee(this.form, this.handleInput, this.handleImageChange.bind(this)),
+      () => renderBody(this.form, this.handleInput, this.handleImageChange.bind(this)),
+      () => renderCard(this.form, this.handleInput, this.handleImageChange.bind(this)),
+      () => renderCaas(this.form, this.handleInput, {
+        contentTypeOptions: this.contentTypeOptions,
+        primaryProductOptions: this.primaryProductOptions,
+      }),
+      () => renderSeo(this.form, this.handleInput),
+      () => renderExperienceFragment(this.form, this.handleInput),
+      () => renderAssetDelivery(this.form, this.handleInput),
+      () => renderUrl(this.form, this.handleInput),
+    ];
+  }
```

Comment:
We may consider wanting to find ways to break some of this up and componentize. Theoretically, we ought to be able to compose this as html if we define the components in their own files, something like:

```
render() {
    return html`
      <h1>Landing Page Builder</h1>
      <div class="builder-container">
          <form>
              <marquee></marquee>
              <marketo></marketo>
              <body></body>
              ...
```